perception/traffic_light_classifier/src/nodelet.cpp

```diff
@@ -94,19 +96,44 @@ void TrafficLightClassifierNodelet::imageRoiCallback(
   output_msg.signals.resize(input_rois_msg->rois.size());
 
   std::vector<cv::Mat> images;
+  std::vector<size_t> backlight_indices;
   for (size_t i = 0; i < input_rois_msg->rois.size(); i++) {
     output_msg.signals[i].traffic_light_id = input_rois_msg->rois.at(i).traffic_light_id;
     const sensor_msgs::msg::RegionOfInterest & roi = input_rois_msg->rois.at(i).roi;
-    images.emplace_back(cv_ptr->image, cv::Rect(roi.x_offset, roi.y_offset, roi.width, roi.height));
+    auto roi_img = cv_ptr->image(cv::Rect(roi.x_offset, roi.y_offset, roi.width, roi.height));
+    if (is_harsh_backlight(roi_img)) {
+      backlight_indices.emplace_back(i);
+    }
+    images.emplace_back(roi_img);
   }
   if (!classifier_ptr_->getTrafficSignals(images, output_msg)) {
     RCLCPP_ERROR(this->get_logger(), "failed classify image, abort callback");
     return;
   }
+
+  for (const auto & idx : backlight_indices) {
+    auto & elements = output_msg.signals.at(idx).elements;
+    for (auto & element : elements) {
+      element.color = tier4_perception_msgs::msg::TrafficLightElement::UNKNOWN;
+      element.shape = tier4_perception_msgs::msg::TrafficLightElement::UNKNOWN;
+    }
```

Comment:
[Q] How should we handle the confidence value here?

Reply:
I think our one option is set the confidence to `0` in those hard situations.

Reply:
The output of tlr classfier is used [here](https://github.com/autowarefoundation/autoware.universe/blob/main/perception/traffic_light_multi_camera_fusion/src/node.cpp#L243-L324). Could you please look at this?

Reply:
@miursh I read the suggested code and confirmed there is no part comparing confidence. Therefore, I updated to set the overwritten confidence to `0.0`, and updated README as well by  https://github.com/autowarefoundation/autoware.universe/pull/5852/commits/51639ecfda940462c330c5872bfcc5b05bd2e5c6